### PR TITLE
watch_later: Persist external `sub_files` and add it as a default option

### PR DESCRIPTION
**Pull request by 0b01**
This pull request enhances support for external subtitle tracks in watch later functionality and adds related integration tests. The main changes include persisting external subtitle files when saving playback state and updating the options and test suite to reflect this capability.

When `save-position-on-quit` is enabled, mpv will save external subtitles and load when the video is reopened.

### CODE REVIEW

**Comment by 0b01** [comment]
Happy to remove `sub-files` from the default list due to privacy concerns.

File: player/configfiles.c [hunk 1]
```diff
@@ -338,11 +338,53 @@ void mp_write_watch_later_conf(struct MPContext *mpctx)
     char **watch_later_options = mpctx->opts->watch_later_options;
     for (int i = 0; watch_later_options && watch_later_options[i]; i++) {
         char *pname = watch_later_options[i];
+
+        // Skip duplicate option names (e.g., from -add on an existing default)
+        bool duplicate = false;
```
Comment (0b01) [inline player/configfiles.c:343]: If `sub-file` is `-add`ed twice then it will be written twice. The list of watch later options should be deduplicated.

**Comment by CounterPillow** [comment]
we gotta move away from GitHub man, now even GitHub's own employees are spamming us with slop.

Look at this git history. 4/6 commits are completely pointless. One is a random merge of some copilot shit. Another deletes some copilot shit that was added in this PR. One is an empty commit that just says "Trigger build" because apparently knowing how to build locally, or that the CI builds on push, is beyond the clanker's knowledge.

And that's before I even look at the code, which I won't do because I'm not your reverse centaur to externalise your 10xness onto.

Oh yeah also the random "test suite" added doesn't even fucking build because it's trying to overflow destination buffers.

**Comment by llyyr** [comment]
Please read https://github.com/mpv-player/mpv/blob/master/DOCS/contribute.md

Getting AI slop contributions that make no sense and contain nonsensical changes from Github employees is not a good look

**Comment by 0b01** [comment]
Sorry was just trying to add a Mac build in order to test my changes locally but should have kept the commit history clean.
